src/api/vdom.js

```diff
@@ -94,8 +94,13 @@ attributes[symbols.default] = function (elem, name, value) {
     value = `[slot="${value}"]`;
   }
 
-  // Set defined props on the element directly.
-  if (name in elem) {
+  // Set defined props on the element directly. This ensures properties like
+  // "value" on <input> elements get set correctly. Setting those as attributes
+  // doesn't always work and setting props is faster than attributes.
+  //
+  // However, certain props on SVG elements are readonly and error when you try
+  // to set them.
+  if (name in elem && !('ownerSVGElement' in elem)) {
```

Comment:
@nadiam84 it's not an own property. You'd have to walk the prototype chain depending on how far up `SVGElement` is.

@bqx You'd have to do `Object.getOwnPropertyDescriptor()` and check `writable`, but just like `hasOwnProperty()` you'd have to do it at the prototype level as it wouldn't be an own property on the element instance.

Depending on the element, how far up you'd have to walk can vary and walking prototype chains can be expensive. Even so, this is performance critical code. The less object accessing and function calls the better.